### PR TITLE
upgrade cryptography python package

### PR DESCRIPTION
**Pull request by cliveseldon**
Update `cryptography` version range

By itself causes build to fail. So need to add a set of further updates:

Python builder updates:
- python builder to 3.8
- ensure grpcio-tools is installed after conda env is created as its used in initial proto compile in tests
- limit protobuf version

Its unclear to me if a subset of these could be achievable. Seems centred on proto compile in python 3.8.

### CODE REVIEW

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 6321322fe85a1a9b02b7acefdeba4035769bc31a | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4540-integration-7t8h9) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 6321322fe85a1a9b02b7acefdeba4035769bc31a | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4540-integration-cp7kx) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: python/setup.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
         "requests<3.0.0",
         "numpy<2.0.0",
         "flatbuffers<2.0.0",
-        "protobuf<4.0.0",
+        "protobuf==3.20.3",
```
Comment (RafalSkolasinski) [inline python/setup.py:30]: That will put hard constraint on `protobuf` version in the user environment causing potential conflicts with other packages.

Comment (adriangonz) [reply python/setup.py:30]: Are both the change in the `protobuf` version and the change in `importlib_metadata` required? I tried locally without those and seems the main Python tests are happy (haven't tried with anything else though).

**Comment by adriangonz** [comment]
I just had a look at the changes and noticed that the Python build image is still on Python 3.7 - so perhaps there's no need to change the Python version?

**Comment by adriangonz** [comment]
After having a deeper look, I think @cliveseldon you were 100% right on adding the extra dep changes. It seems something to do with `tensorflow` forcing an older `protobuf`, and also with running an old version of `flake8` - but neither of them would be trivial fixes.

@RafalSkolasinski if you can have a look at the latest changes, it would be great to get your thumbs up.

**Comment by RafalSkolasinski** [comment]
Looks good - shall we fire integration & notebook tests?

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 71d1d605074b4188c6bad90735f8a1633e3fb07e | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4540-integration-wmv8c) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 71d1d605074b4188c6bad90735f8a1633e3fb07e | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4540-integration-ppz8g) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 71d1d605074b4188c6bad90735f8a1633e3fb07e | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4540-integration-ppz8g) | `/test integration`
notebooks | 71d1d605074b4188c6bad90735f8a1633e3fb07e | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4540-notebooks-r5xgh) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 71d1d605074b4188c6bad90735f8a1633e3fb07e | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4540-notebooks-r5xgh) | `/test notebooks`
integration | 71d1d605074b4188c6bad90735f8a1633e3fb07e | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4540-integration-lvvbz) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 71d1d605074b4188c6bad90735f8a1633e3fb07e | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4540-notebooks-qrlg5) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by adriangonz** [comment]
Just had a look at the notebook tests, and they seem to fail because of an issue not related to this PR (which should get fixed in #4691 ), so I'll go ahead and merge this one.

**Comment by mwm5945** [comment]
@cliveseldon @adriangonz any chance this would be part of a 1.14.2 release?

**Comment by adriangonz** [comment]
Hey @mwm5945 ,

We don't have plans atm to backport the fix to `1.14.x`. Would upgrading to `1.15.x` be an option for you?

**Comment by mwm5945** [comment]
@adriangonz potentially, though a 1.14 release would be preferred (people get anxious with package upgrades haha). Is there a large difference between 14 and 15?

**Pull request merged**